packages/material-ui/src/SelectWidget/SelectWidget.tsx

```diff
@@ -75,7 +75,6 @@ const SelectWidget = ({
     <TextField
       id={id}
       label={label || schema.title}
-      name={name}
```

Comment:
It's not clear to me why you would want to drop this?

Reply:
It is a non-declared variable and causes the server-side rendering in nextjs to reject it.

Take a look at the full file, do a search and name didn't appear anywhere else.

https://github.com/rjsf-team/react-jsonschema-form/blob/a1eb74792f0ae46335fc8f2845db03ee7e3f095b/packages/material-ui/src/SelectWidget/SelectWidget.tsx

Reply:
I'm wondering if it would be better to extract this property rather than drop it? Maybe I'm mistaken, but isn't it passed into the constructor for the widget? (Same applies for the other dropped reference.)

Reply:
It would, I assume, be preferable to have name attached to every widget type that renders to a form of input at some point. However, I am not familiar enough with the code to know why that doesn't seem to be the case in this instance. It is possible my search missed something.

![nameSearch](https://user-images.githubusercontent.com/4021807/85088781-8575d180-b1a6-11ea-8bba-0ce578844589.png)

I checked on the playground, and the default theme is not passing through a name, and even with the name included in the material theme it is not actually coming through as a proper name for the input box. As such, I had assumed this was an architectural decision that someone had made, and without knowing more I didn't want to change the code more than is required to fix it.

![defaultTheme](https://user-images.githubusercontent.com/4021807/85089034-295f7d00-b1a7-11ea-9fd0-033e5090cec9.png)
![materialTheme](https://user-images.githubusercontent.com/4021807/85089036-29f81380-b1a7-11ea-9cff-fedd252dc3b0.png)

Reply:
Thanks for being patient with me. :) It looks like the name is used in the default theme if a title is not set. As I poke around more I see what you're seeing and I concur with what you're doing here. I am actually wondering now though, if we should just drop it entirely from the `<TextWidget />` too, rather than add it to the params?

Reply:
It is quite possible that this would be the thing to do. Sorry if I seem short with anything, life has been super busy. I was very happy these tools ready existed or I would have been building them from the ground up... I realized that I have spent ~ 4 years building proprietary "foundational" (in the company I was working at) components and am super thankful that others are doing things that I am now able to use in my own projects.

I realized with the TextWidget commit in the 2.0.0 branch there were some things that may make it difficult if someone is trying to theme some of the parts that I personally am not yet using. Should I be mentioning that somewhere? Thank you for your patience with me.

Reply:
Not sure I follow? I think dropping the name in this one should be fine.

Reply:
I dropped as requested. My question was about this line `<List dense={true} disablePadding={true}>` (and others that directly override a MaterialUI preference) because then it would not be possible to do the overriding of `dense` and `disablePadding` [if someone wanted to with default props](https://material-ui.com/customization/globals/) in their own theme.

```
const theme = createMuiTheme({
  props: {
    List: {
      // The default props to change
      disablePadding: false, // This wouldn't work, as the prop is passed in manually
    },
  },
});
```

I was not requiring it changed but was not sure how generic this theme is meant to be. It wasn't specifically an issue (at least for me) and I am new to the open-source world so I wasn't sure where to put the question appropriately.

Reply:
Ahhhh, gotcha, I'm tracking now. Great question. The use of <List/> here is an alternative to MUI's typical use of the helper text for errors. The addition of dense and disablePadding were an attempt to match the indentation and alignment of helper text.